### ISSUE TITLE
[macro] `.macro.hx` not working for build macros

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#8159]
It seems like this should work (and might help dealing with issues such as #8066).

Instead, it fails with `Type not found`:

`Main.hx`

```haxe
@:build(Macro.build()) // Type not found : Macro
class Main {
	static function main() {}
}
```

`Macro.macro.hx`

```haxe
class Macro {
	public static function build() {
		return null;
	}
}
```

**Comment by ncannasse** [comment]
macro is not a platform. I guess it works with .eval.hx ?

**Comment by Gama11** [comment]
No, I already tried that, and that doesn't work either.

**Comment by nadako** [comment]
We also look for `macro` postfix and this is pretty handy to avoid all the wrong typing context issues, and I think this should work too.

**Comment by kevinresol** [comment]
`Macro.hx` has to exist too. Otherwise the compiler will not search for `Macro.macro.hx`.
https://github.com/HaxeFoundation/haxe/blob/91fcc0d142d0af5301f457d82fca9353dff6a01d/src/context/common.ml#L617

**Comment by RealyUniqueName** [comment]
Fixed by #8202
